lib/modules/dosomething/dosomething_reportback/dosomething_reportback.query.inc

```diff
@@ -123,25 +121,26 @@ function dosomething_reportback_get_reportbacks_query($params) {
   return $result;
 }
 
-
 /**
- * Filter out requested flagged Reportbacks based on user permissions.
- *
- * @param  array  $params
- * @return mixed
+ * Determine if user has access to flagged Reportbacks based on user permission. 
+ * @param array $results
  */
-function dosomething_reportback_filter_flagged_reportbacks($params) {
+function dosomething_reportback_accessible_results ($results) {
   if (user_access('view any reportback')) {
-    if (isset($params['flagged'])) {
-      unset($params['flagged']);
+    return $results;
+  }
+  else {
+    foreach ($results as $key => $reportback) {
+      if ($reportback->flagged == 1) {
```

Comment:
@weerd actually, probably more helpful to have that comment here!  

The `===` was breaking the code because it was also checking the type to see if it was a boolean. Since `$reportback->flagged` is a string, it returned false and never unset.

Would it be better to use `==` in this case or still use `===` and change this to 
`if ($reportback->flagged === "1") {`?